WordPress/WordPressTest/ContextManagerMock.swift

```diff
@@ -2,7 +2,27 @@ import XCTest
 
 @testable import WordPress
 
-extension ContextManagerMock {
+class ContextManagerMock: ContextManager {
+
+    /// Create an in memory database using the current version of the data model.
+    convenience init() {
+        self.init(modelName: ContextManagerModelNameCurrent)
+    }
+
+    /// Create an in memory database using given `modelName`.
+    ///
+    /// - SeeAlso `ContextManager`
+    init(modelName: String) {
+        super.init(modelName: modelName, store: URL(fileURLWithPath: "/dev/null"))
+    }
+
+    override func saveContextAndWait(_ context: NSManagedObjectContext) {
+        super.saveContextAndWait(context)
+        // This log magically resolves a deadlock in
+        // `ZDashboardCardTests.testShouldNotShowQuickStartIfDefaultSectionIsSiteMenu`
```

Comment:
Pedantic / nitpick

```suggestion
        // This log resolves a deadlock we previously experienced in
        // `DashboardCardTests.testShouldNotShowQuickStartIfDefaultSectionIsSiteMenu`
        //
        // Unfortunately, we're not sure why.
```

- Removed the `Z` prefix because of https://github.com/wordpress-mobile/WordPress-iOS/pull/18660 (even though that hasn't been merged yet, I think we can live with the temporary inconsistency in the interest of closing the loop here)
- This might just be me being pedantic, but I don't like the the use of "magic". There should be no magic in what we do. Pragmatically, it's okay to admit that we don't understand why something happens, _yet_, and that we decided not to invest additional time into it in the interest of tackling more impactful tasks.

To acknowledge the fact that I might be overly pedantic with my "magic avoidance", here's another suggestions just for the `Z` removal. Feel free to apply the one you prefer, no hard feelings 😄 

```suggestion
        // This log magically resolves a deadlock in
        // `DashboardCardTests.testShouldNotShowQuickStartIfDefaultSectionIsSiteMenu`
```

Reply:
Related: The previous implementation had a `FIXME` comment.


https://github.com/wordpress-mobile/WordPress-iOS/blob/47fef86327e91cd1decbcab04d32a5c84fdfb344/WordPress/WordPressTest/ContextManagerMock.m#L13

Did you intentionally remove it, or was it "lost in translation"?

I initially thought it would be okay to remove it. But, while writing my comment in favor for it, I realized that it would actually be good to have some semi-automated reminder for this. We have lots of work left to do in the overall Core Data area, so it's possible this will sort itself out, but if it doesn't, there might be something going on underneath that we might want to investigate at some point.

What do you think? An alternative could be to file a reminder task along the lines of "check if the `NSLog` in `ContextManagerMock` is still needed after every Core Data improvement iteration."

Reply:
Good catch! It's indeed a lost in translation. I'll add it back.

Reply:
All nitpicks are welcomed and appreciated! I've [taken the wording](https://github.com/wordpress-mobile/WordPress-iOS/pull/18655/commits/b1ec03aa7e6010ae0d50bbfbfcc419ffc9b2a652) in your suggestion. Thanks! 👍